src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -538,7 +538,8 @@ def _check_subnet(s):
 def _subnet_capacity_check(subnet_mask, vmss_instance_count):
     mask = int(subnet_mask)
     # '2' are the reserved broadcasting addresses
-    return ((1 << (32 - mask)) - 2) > vmss_instance_count
+    # '*1.5' so we have enough leeway for over-provision
+    return ((1 << (32 - mask)) - 2) > int(vmss_instance_count * 1.5)
```

Comment:
Shouldn't this only apply if the overprovision setting is enabled then?  If I don't want overprovision why should I arbitrarily be prevented from using a subnet that will fit everything?

Reply:
I thought about it but was not so sure whether it is worthwhile to get to that accurate, but since you raise it, i will do it.